workshop/mathIsFun/ComplexNumber.cpp

```diff
@@ -0,0 +1,165 @@
+#include "ComplexNumber.h"
+#include <string>
+
+/*
+ * default constructor, initialize number to be zero
+*/
+ComplexNumber::ComplexNumber() :
+    m_real(0), 
+    m_imaginary(0)
+{}
+
+
+/*
+ * gets real and imaginary number for new complex number
+ * @param real real number
+ * @imaginary imaginary number
+*/
+ComplexNumber::ComplexNumber(const double real, const double imaginary) :
+    m_real(real),
+    m_imaginary(imaginary)
+{}
+
+
+/*
+ * getter for the real number part
+ * @return the real number part from the complex number
+*/
+double ComplexNumber::getReal() const noexcept {
+    return m_real;
+}
+
+
+/*
+ * setter for the real number part
+ * @parram real new real number part of the complex number
+*/
+void ComplexNumber::setReal(const double real) noexcept {
+    m_real = real;
+}
+
+
+/*
+ * getter for the imaginary number part
+ * @return the imaginary number part from the complex number
+*/
+double ComplexNumber::getImaginary() const noexcept {
+    return m_imaginary;
+}
+
+
+/*
+ * setter for the imaginary number part
+ * @parram imaginary new imaginary number part of the complex number
+*/
+void ComplexNumber::setImaginary(const double imaginary) noexcept {
+    m_imaginary = imaginary;
+}
+
+
+/*
+ * checks if complex numbers are equal
+ * @param other complex number to be compared with this
+ * @return true if both numbers are equal else false
+*/
+bool ComplexNumber::equal(const ComplexNumber& other) const noexcept {
+    return (this->m_real == other.m_real) && (this->m_imaginary == other.m_imaginary);
+}
+
+
+/*
+ * sum two complex numbers
+ * @param other complex number to be sum with this
+ * @return sum of this and other
+*/
+ComplexNumber ComplexNumber::add(const ComplexNumber& other) const noexcept {
```

Comment:
What does these private methods add to your implementation?
How are they better than just implementing the logic directly in the overloaded operator functions?
This isn't really all that bad, just seem like alot of extra unneeded work.